### PR TITLE
Add AppVeyor CI files

### PR DESCRIPTION
**Pull request by lgpage**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  CMD_IN_ENV: "cmd /E:ON /V:ON /C obvci_appveyor_python_build_env.cmd"
+
+  # Workaround for https://github.com/conda/conda-build/issues/636
+  PYTHONIOENCODING: "UTF-8"
+
+  matrix:
+    - TARGET_ARCH: "x64"
+      CONDA_PY: "34"
+      PY_CONDITION: "python >=3.4,<3.5"
+    - TARGET_ARCH: "x64"
```
Comment (jhamrick) [inline appveyor.yml]: Does python 3.6 fail to build? Do you know why?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    - cmd: conda update --yes --quiet conda
+    - cmd: set PYTHONUNBUFFERED=1
+    - cmd: conda config --add channels conda-forge
+    - cmd: conda info
+    - cmd: conda create -n nbgrader-dev "%PY_CONDITION%"
+    - cmd: conda env update
+    - cmd: activate nbgrader-dev
+
+    # Install non-conda packages
+    - cmd: pip install pytest-rerunfailures
```
Comment (jhamrick) [inline appveyor.yml]: Rather than installing this explicitly maybe we can do `pip install -r dev-requirements-windows.txt` below?

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: nbgrader-dev
```
Comment (jhamrick) [inline environment.yml]: Hmm, I'm tempted to say I'd prefer to try to install everything via pip to make sure that that method of installation still works on windows (I'm more confident that conda will work reliably than pip, TBH). And then we don't have to maintain two separate files with dependencies.

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    - cmd: npm install phantomjs-prebuilt
+    - cmd: set "PATH=%CD%\\node_modules\\phantomjs-prebuilt\\lib\\phantom\\bin;%PATH%"
+
+# Skip .NET project specific build phase.
+build: off
+
+test_script:
+    - cmd: python setup.py develop
```
Comment (jhamrick) [inline appveyor.yml]: This line could be replaced with `pip install -r dev-requirements-windows.txt -e .` which will install all regular dependencies and dev dependencies and install the package itself.

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    - cmd: jupyter nbextension install --sys-prefix --py nbgrader
```
Comment (jhamrick) [inline appveyor.yml]: These lines shouldn't be necessary, *I think*. In the tests I believe temporary directories are created to test the nbextension installation.

File: appveyor.yml (hunk 1 above)
Comment (lgpage) [inline appveyor.yml]: Yes python 3.6 fails to build, at least with the conda env approach I have taken here, as there are a huge amount of missing python 3.6 conda dependencies. They are in the process of re-rendering current feedstocks to get python 3.6 packages out so it would be long until this can be added.

File: appveyor.yml (hunk 2 above)
Comment (lgpage) [inline appveyor.yml]: Can do. I did this as it is the only package needed for the tests :smile:

File: environment.yml (hunk 3 above)
Comment (lgpage) [inline environment.yml]: Either way, If we take a ``pip`` approach or ``conda`` approach I would really like to have this file in the repo as it would make it easier to setup a fresh conda dev env for nbgrader.

**Comment by lgpage** [comment]
@jhamrick I followed a ``conda`` env approach here, as it was easy to copy and paste from what I had lying around :wink: and I know for previous experience AppVeyor can sometimes be a bitch when it comes to missing system packages not installed when using ``pip``.

**Comment by lgpage** [comment]
In any case I'll play around with a ``pip`` approach version tomorrow.

File: appveyor.yml (hunk 1 above)
Comment (jhamrick) [inline appveyor.yml]: Ah, I see. Yeah, I would say let's build with pip then if that's the case, as I'd ideally like to keep the 3.6 builds.

File: environment.yml (hunk 3 above)
Comment (jhamrick) [inline environment.yml]: That's true. How about not including it for this PR, and then make another PR with an environment file and then also update the dev installation docs telling people how they can create a new conda dev env with it?

**Comment by jhamrick** [comment]
> AppVeyor can sometimes be a bitch when it comes to missing system packages not installed when using pip

Ah, I see. Hmm. Well, let's try it with pip, and if you can't get it to work then maybe we can use conda but it would be great if people were able to install with pip. That's how I was installing it on Jenkins, so it *should* be able to work.

File: appveyor.yml (hunk 1 above)
Comment (lgpage) [inline appveyor.yml]: Do you also want x86 tests?

Comment (jhamrick) [inline appveyor.yml]: No, I think x64 is fine.

**Comment by lgpage** [comment]
@jhamrick please kill this build: https://ci.appveyor.com/project/jupyter/nbgrader/build/1.0.4

**Comment by jhamrick** [comment]
Done. Do you not have permission to kill/restart builds yourself? Let me see if I can fix that...

**Comment by lgpage** [comment]
I don't, thanks :smile:

**Comment by jhamrick** [comment]
OK, try now?

**Comment by lgpage** [comment]
Nope, still not.

**Comment by jhamrick** [comment]
Hmm. Are you logged in with GitHub, or with a username and password? I made you an admin on the repo, so if you're logged in with GitHub, it should work...

**Comment by lgpage** [comment]
I'm logging in with GitHub. Restarted my browser, clear cache and cookies and still not :cry:

**Comment by jhamrick** [comment]
Ok, that was confusing and complicated. I think I figured it out now -- I figured access would just be inherited from GitHub (like is the case on Travis) but not so. Let me know if it still isn't working...

**Comment by lgpage** [comment]
I have access now thanks :D and the tests should be passing. I hate phantomjs !! Using the AppVeyor system version leads to that completely unrelated failure occurring with the fixture finalizer :angry:

**Comment by lgpage** [comment]
@jhamrick well the tests are passing, sort of... It looks like we will have to fix the underlying IOPub TimeOut issue for this to be useful. At the moment it is failing more often than not due to this issue.

**Comment by jhamrick** [comment]
This is really, awesome, thanks for getting this working! I am going to go ahead and merge it despite the appveyor tests not fully passing, and we can try to address the IOPub issue in a separate PR. Pretty annoying that that's happening, but I guess some slow continuous integration that causes sporadic issues is better than no continuous integration at all 😆

**Comment by lgpage** [comment]
The best would be to post a PR for ``nbconvert`` to make the iopub timeout configurable instead of hard coded here https://github.com/jupyter/nbconvert/blob/master/nbconvert/preprocessors/execute.py#L278.

**Comment by jhamrick** [comment]
Yeah, good idea, I'll do that.
